### DISCUSSION TITLE
{object}  is an object but has no label key error

### DISCUSSION THREAD

**Discussion by igorlanko** [janosh/svelte-multiselect#261] [Q&A] [answered]
Getting a console error when just getting started with the component:
```
{object} is an object but has no label key 
```
As I understand it, an array passed into `options` prop must have `label` and `key` in it. However I'm pulling data from a Supabase table that's setup with different names.

How can I tell the component to use different prop names for `label` and `key`? Or what would be a better approach? 

Thanks!

<img width="610" alt="CleanShot 2023-09-11 at 17 57 28@2x" src="https://github.com/janosh/svelte-multiselect/assets/31314207/fd2e60d1-7b18-4934-bae6-14b3a69b4eba">

_The amount of errors is ridiculous, lol_

**Comment by igorlanko** [comment]
Mapping does the trick, but I was wondering if there's a prop I'm on the parent I'm missing 🤷 
```ts
$: allTags = data.allTags.map((tag) => {
   return { label: tag.title, value: tag.title }
})
```

**Reply by janosh** [reply, marked as answer]
Yep, that's the recommend solution. The `value` key is optional and defaults to `label` if missing. So you could do

```svelte
$: allTags = data.allTags.map((tag) => ({ label: tag.title }))
```

**Reply by igorlanko** [reply]
Thanks!
